### PR TITLE
Consider dispatch from `useActionState` stable

### PR DESCRIPTION
**Pull request by eps1lon**
## Summary

The dispatch function from `useState` and `useReducer` is stable so `exhaustive-deps` doesn't require them to be in dependencies. This applies the same rule to `useActionState` and `useFormState`

## How did you test this change?

- `yarn test packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js` fails with on the first commit but passes on the second.

### CODE REVIEW

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js [hunk 1]
```diff
@@ -179,6 +179,10 @@ export default {
       //               ^^^ true for this reference
       // const [state, dispatch] = useReducer() / React.useReducer()
       //               ^^^ true for this reference
+      // const [state, dispatch] = useReducer() / React.useActionState()
```
Comment (rachitiitr) [inline packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js:182]: Typo? I think you meant - `useActionState() / React.useActionState()`

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js [hunk 2]
```diff
@@ -179,6 +179,10 @@ export default {
...
+      //               ^^^ true for this reference
+      // const [state, dispatch] = useReducer() / ReactDOM.useFormState()
```
Comment (rachitiitr) [inline packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js:184]: Typo? I think you meant - `useFormState() / ReactDOM.useFormState()`

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js [hunk 3]
```diff
@@ -260,7 +265,12 @@ export default {
           }
           // useEffectEvent() return value is always unstable.
           return true;
-        } else if (name === 'useState' || name === 'useReducer') {
+        } else if (
+          name === 'useState' ||
+          name === 'useReducer' ||
+          name === 'useFormState' ||
+          name === 'useActionState'
```
Comment (rachitiitr) [inline packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js:272]: Would we want to shorten this to - 
```
['useState', 'useReducer', 'useFormState', 'useActionState'].includes(name)
```

Comment (eps1lon) [reply packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js:272]: One less array allocation and can be better optimized by runtimes.

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js (hunk 1 above)
Comment (eps1lon) [reply packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js]: Thanks, fixed.

File: packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js (hunk 2 above)
Comment (eps1lon) [reply packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js]: Thanks, fixed.

**Comment by sophiebits** [comment]
Should we just do this logic for the new name and not the old one?

**Comment by eps1lon** [comment]
It's already out there with the lightest warning that it's just been renamed i.e. you can still use `ReactDOM.useFormState`. I don't see a harm treating `useFormState` just like `useActionState` because that's what it ultimately is. If it would require more implementation effort to add support for it, I'd probably use your argument but since it's pretty trivial to implement, I don't see an issue.

**Comment by eps1lon** [comment]
Going with just `useActionState` to align with the React Compiler.

**Comment by eps1lon** [comment]
Aligned with React Compiler to only consider `React.useActionState`. Code using `ReactDOM.useFormState` should just switch to `React.useActionState`

**Review by gaearon — APPROVED** [review]
